### PR TITLE
CI matrix: Add windows 20H2 - fix TLS1.3 issue

### PR DESCRIPTION
**Pull request by aik-jahoda**
When employ Windows server 20H2 into CI we got buch of TLS 1.3 errors:

> System.ComponentModel.Win32Exception : The client and server cannot communicate, because they do not possess a common algorithm.

Usually it means the server/client doesn't support TLS 1.3

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-revert-62098-c4c66b16e5dd4304a8/System.Net.Security.Tests/1/console.77465e3d.log?%3F%253Fsv%253D2019-07-07%2526se%253D2021-12-21T09%25253A56%25253A18Z%2526sr%253Dc%2526sp%253Drl%2526sig%253D6ptrdfy%25252FY91GLeQ4bm8kc%25252FtyN0EXeNccWwtsbnhIAIw%25253D

replace #60054

part of #57947

### CODE REVIEW

**Comment by aik-jahoda** [comment]
Based on https://docs.microsoft.com/en-us/windows/win32/secauthn/protocols-in-tls-ssl--schannel-ssp- the TLS 1.3 is not supported (not explicitly, but there is no support for Windows 20H2)

Probably there is an issue with Windows version check.
/cc @wfurt

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 1]
```diff
@@ -431,6 +432,8 @@ private static bool GetTls13Support()
                 {
                     client = Registry.GetValue(clientKey, "Enabled", null);
                     server = Registry.GetValue(serverKey, "Enabled", null);
+                    Console.WriteLine($"client {client}");
+                    Console.WriteLine($"client {client}");
```
Comment (wfurt) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:436]: Should be server?

**Comment by wfurt** [comment]
The support check will respect registry. Based on the output somebody/something enabled TLS 1.3 even if not officially supported. (The code was there for a while just not enabled by default)
My recommendation would be to find what enables it and remove that code.

**Comment by wfurt** [comment]
BTW this may go back to base Helix image. I don't know who and how created that. 
cc: @MattGal

**Comment by wfurt** [comment]
Thanks for quick update @MattGal. 
It seems like from the debug and side conversion with @aik-jahoda KEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client\Enabled and KEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server\Enabled are set , forcing TLS 1.3 on platform that officially does not support it. 

Can we either flip them to 0 or simply leave them empty?

**Comment by aik-jahoda** [comment]
The Windows version is 10.0.19042 , however the check in Platform detection https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L413
assumes the TLS 1.3 is available from 10.0.19041. 

@wfurt, shouldn't be safer to change the IsWindows10Version2004OrGreater (20H1) to `is 20H2 or greater` (10.0.19044)? It would prevent this error when the registry are set but the Windows doesn't support it

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs (hunk 1 above)
Comment (aik-jahoda) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:436]: Yeah, it served its purpose for getting the CI machine info. Reverting...

**Comment by aik-jahoda** [comment]
New helix machine will be available 12/15

**Comment by aik-jahoda** [comment]
Tests: https://dev.azure.com/dnceng/public/_build/results?buildId=1514902&view=logs&j=da1be4e8-11c0-5569-f8d4-0fcb9670c66a

There are unrelated osx/linux failures

**Review by danmoseley — APPROVED** [review]

**Comment by ericstj** [comment]
@safern do we need to rerun this to make sure the test results are still OK from outerloop or is it OK to merge?

**Comment by wfurt** [comment]
The failing tests were innerloop  in the past. We can run outer loop but I don't think it is necessary.

**Comment by danmoseley** [comment]
No harm in re-running at least the regular checks since it's been a while..

**Comment by ericstj** [comment]
Closing this since the PR is now stale and the overall change is tracked by https://github.com/dotnet/runtime/issues/57947

**Pull request closed without merging**